lightning/src/util/ser_macros.rs

```diff
@@ -175,7 +175,8 @@ macro_rules! decode_tlv {
 }
 
 macro_rules! decode_tlv_stream {
-	($stream: expr, {$(($type: expr, $field: ident, $fieldty: tt)),* $(,)*}) => { {
+		($stream: expr, {$(($type: expr, $field: ident, $fieldty: tt)),* $(,)*}
+		 $(, ($custom_type: tt, $custom_value: ident, $custom_range_min: expr, $custom_range_max: expr))?) => { {
```

Comment:
Catching up with the conv, IIUC each custom TLV inside the onion would have min/max bounds defined at the protocol-level ? Otherwise if DoS protections are per-implementation, you might fail deser of a message gauged by valid as the sending implementation. I still have in mind the use-case of sending Lightning transactions inside onions, and a commitment transaction size can dynamic on the number of HTLC outputs.

Reply:
>  IIUC each custom TLV inside the onion would have min/max bounds defined at the protocol-level ?

I don't think the bounds are spec'd for custom TLVs, but yup. 

> I still have in mind the use-case of sending Lightning transactions inside onions, and a commitment transaction size can dynamic on the number of HTLC outputs.

Cool, that should be supported. Currently we support reading 1 TLV within the type range `64..u64::max_value()` (can be an invoice_request, etc or custom). Lmk if there's a change desired here.

Reply:
> IIUC each custom TLV inside the onion would have min/max bounds defined at the protocol-level ? 
IIUC the closest spec is that the generalized TLV format defines that types between 2^16 and u64::max_value() are reserved for custom types.
https://github.com/lightning/bolts/blob/16973e2b857e853308cafd59e42fa830d75b1642/01-messaging.md?plain=1#L131